disco_aws_automation/disco_vpc_peerings.py

```diff
@@ -225,143 +232,171 @@ def _get_peering_lines():
+                self.client.describe_vpc_peering_connections,
                 Filters=create_filters({'accepter-vpc-info.vpc-id': [vpc_id]})
             )['VpcPeeringConnections']
         else:
-            peerings = throttled_call(client.describe_vpc_peering_connections)['VpcPeeringConnections']
+            peerings = throttled_call(self.client.describe_vpc_peering_connections)['VpcPeeringConnections']
 
         peering_states = LIVE_PEERING_STATES + (["failed"] if include_failed else [])
         return [
             peering
             for peering in peerings
             if peering['Status']['Code'] in peering_states
         ]
+
+    def _resolve_peering_connection_line(self, line):
+        """
+        Resolve a peering connection line into a set of PeeringConnections. Expand any wildcards
+
+        Args:
+            line (str): A peering line like `vpc_name[:vpc_type]/metanetwork vpc_name[:vpc_type]/metanetwork`
+                        `vpc_name` may be the name of a VPC or a `*` wildcard to peer with any VPC of vpc_type
+        """
+
+        # convert the config line into a PeeringConnection but it may contain wildcards
+        unresolved_peering = PeeringConnection.from_peering_line(line)
+
+        # get all VPCs created through Asiaq. Ones that have type and Name tags
+        existing_vpcs = [vpc for vpc in throttled_call(self.client.describe_vpcs).get('Vpcs', [])
+                         if all(tag in tag2dict(vpc.get('Tags', [])) for tag in ['type', 'Name'])]
+
+        if '*' in (unresolved_peering.source_endpoint.type, unresolved_peering.target_endpoint.type):
+            raise VPCConfigError('Wildcards are not allowed for VPC type in "%s". '
+                                 'Please specify a VPC type when using a wild card for the VPC name' % line)
+
+        def resolve_endpoint(endpoint):
+            """
+            Convert a PeeringEndpoint that may contain wildcards into a list of PeeringEndpoints
+            with wildcards resolved
+            """
+            endpoints = []
+            for vpc in existing_vpcs:
+                tags = tag2dict(vpc['Tags'])
+                if endpoint.name in ('*', tags['Name']) and endpoint.type == tags['type']:
+                    endpoints.append(PeeringEndpoint(tags['Name'], tags['type'], endpoint.metanetwork, vpc))
+            return endpoints
+
+        # find the VPCs that match the peering config. Replace wildcards with real VPC names
+        source_endpoints = resolve_endpoint(unresolved_peering.source_endpoint)
+        target_endpoints = resolve_endpoint(unresolved_peering.target_endpoint)
+
+        # generate new connection lines by peering the cross product of every source and target endpoint
+        return {PeeringConnection(source, target)
+                for source, target in product(source_endpoints, target_endpoints)
+                # Don't peer a VPC with itself
+                if not source == target}
```

Comment:
Might be a little more concise to exclude source endpoints from the target endpoints via a set difference (have resolve return a set instead of a list).